### PR TITLE
Radio and Bluetooth Abstraction!!  📻 🟦🦷

### PR DESCRIPTION
**Pull request by JeremyButt**
abstracting away from which radio lib is being used! 🗂
This abstraction layer helps us develop code in the simulator that will seemlessly map to libraries for the physical device. 

Ask if you have any other questions!

### CODE REVIEW

File: radio/rfm69/RFM69RxTx.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+class RFM69RxTx: public ChirpRxTx
+{
+    public:
+        RFM69RxTx();
+        void init();
+        void resetRadio();
+        void send(int, char[], size_t);
+        char* receive();
+        void sendBluetooth(char[], size_t);
```
Comment (s-axford) [inline radio/rfm69/RFM69RxTx.h]: Bluetooth is handled by nrf52 not rfm69

Comment (JeremyButt) [reply radio/rfm69/RFM69RxTx.h]: Made Separation

**Comment by s-axford** [comment]
I think the bluetooth interface files should be in the bluetooth directory, not the radio directory. I will change it and commit to this branch if nobody objects

**Comment by s-axford** [comment]
I went through and changed the code so that the bluetooth stuff is in the bluetooth directory. then I changed the cpp code so that it loads onto the board and works as before.

I suggest @JeremyButt reviews my changes before merging to ensure it all still matches the simulator interfaces. (I had to change the bluetooth interfaces as size_t does not exist for Ardiuno)

I also suggest @DelightedWaif and @johnnyb8  review what I did to the bluetooth directory, the interface is now in a sketch folder called nrf52, I have moved the working bluetooth code that you both wrote into the sketch folder and named it nrf52 to match the sketch folder name. I also moved the bluetooth code into the interfaces. I believe this is working as before, but may need some testing

**Review by s-axford — DISMISSED** [review]
I suggest @JeremyButt reviews my changes to the interfaces 

@DelightedWaif and @johnnyb8 should review my changes to the bluetooth directory

File: bluetooth/nrf52/NRF52RxTx.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#include <bluefruit.h>
+#include <SPI.h>                // Included with Arduino IDE, should also add https://github.com/LowPowerLab/SPIFlash
+#include "ChirpBluetoothRxTx.h"
+
+// Function prototypes for packetparser.cpp
+uint8_t readPacket (BLEUart *ble_uart, uint16_t timeout);
+float   parsefloat (uint8_t *buffer);
+void    printHex   (const uint8_t * data, const uint32_t numBytes);
```
Comment (JeremyButt) [inline bluetooth/nrf52/NRF52RxTx.h]: packetparser.cpp is not used remove this.

File: bluetooth/nrf52/NRF52RxTx.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+// Packet buffer
+extern uint8_t packetbuffer[];
```
Comment (JeremyButt) [inline bluetooth/nrf52/NRF52RxTx.h]: Remove this.

File: bluetooth/nrf52/NRF52RxTx.cpp [hunk 4]
```diff
@@ -0,0 +1,91 @@
+#include "NRF52RxTx.h"
+
+// OTA DFU service
+BLEDfu bledfu;
+
+// Uart over BLE service
+BLEUart bleuart;
+
+void startAdv(void)
```
Comment (JeremyButt) [inline bluetooth/nrf52/NRF52RxTx.cpp]: This should defined in the header as a private member funct.

**Comment by TimothyMather** [comment]
Packet Parser is renamed but never used

**Review by TimothyMather — DISMISSED** [review]
Please remove packetparser.cpp

File: bluetooth/nrf52/NRF52RxTx.cpp [hunk 5]
```diff
@@ -0,0 +1,91 @@
+#include "NRF52RxTx.h"
+
+// OTA DFU service
+BLEDfu bledfu;
+
+// Uart over BLE service
+BLEUart bleuart;
```
Comment (JeremyButt) [inline bluetooth/nrf52/NRF52RxTx.cpp]: put this in constructor? @s-axford im going to try this. we'll have to test arduino compiling before merge.

File: bluetooth/nrf52/NRF52RxTx.h [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+class NRF52RxTx: public ChirpBluetoothRxTx
+{
+    public:
+        NRF52RxTx();
+        void init();
+        void send(char[], size_t);
+        char* receive();
```
Comment (JeremyButt) [inline bluetooth/nrf52/NRF52RxTx.h:11]: add private member startAdv

Comment (JeremyButt) [reply bluetooth/nrf52/NRF52RxTx.h:11]: and the private vars for the bleuart and bledfu

File: radio/rfm69/RFM69RxTx.cpp [hunk 7]
```diff
@@ -0,0 +1,67 @@
+#include "RFM69RxTx.h"
+
+RFM69 radio(RF69_SPI_CS, RF69_IRQ_PIN, false, RF69_IRQ_NUM);
```
Comment (JeremyButt) [inline radio/rfm69/RFM69RxTx.cpp]: try to move to constructor. @s-axford testing with board will be required

**Comment by JeremyButt** [comment]
Made changes to the constructors and removed the unused packet parser. @s-axford @DelightedWaif  can you guys test the bluetooth funct and radio funct to see that the constructors still work and that the packet parser removal didnt break the bluetooth.

**Review by s-axford — APPROVED** [review]

**Review by TimothyMather — APPROVED** [review]

**Pull request merged**